### PR TITLE
DC-554, DC-514: remove perf test config, fix dev test config

### PR DESCRIPTION
**Pull request by pshapiro4broad**
Now that the perf environment no longer exists, we can no longer use it for running our nightly tests. As described in DC-554, we will no longer run nightly tests in any environment. Currently the integration tests are run:

- on PRs, and after merge to `main` (via GitHub actions)
- on QA deployment to alpha and staging (via QA automation in Jenkins)

This change

- adds a working `dev` deployment (DC-514)
- renames the nightly test to `integration`
- renames the existing `dev` deployment to `local`, and update scripts / documentation to reflect this
- changes the slack notification to link to the job instead of the workflow

See also https://github.com/broadinstitute/dsp-jenkins/pull/639

Example run of integration test using the `dev` environment: https://github.com/DataBiosphere/terra-data-catalog/actions/runs/3184211014/jobs/5192370900
Example using the `local` environment: https://github.com/DataBiosphere/terra-data-catalog/actions/runs/3182720980/jobs/5189066228

### CODE REVIEW

**Comment by pshapiro4broad** [comment]
Example run of integration test using the `dev` environment: https://github.com/DataBiosphere/terra-data-catalog/actions/runs/3184211014/jobs/5192370900

File: integration/src/main/resources/suites/local/FullIntegration.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "FullIntegration",
+  "description": "All integration tests",
+  "serverSpecificationFile": "catalog-local.json",
```
Comment (s-rubenstein) [inline integration/src/main/resources/suites/local/FullIntegration.json:4]: Where is `catalog-local.json` stored?

**Review by s-rubenstein — APPROVED** [review]
LGTM

File: integration/src/main/resources/suites/local/FullIntegration.json (hunk 1 above)
Comment (pshapiro4broad) [reply integration/src/main/resources/suites/local/FullIntegration.json:4]: It already existed; it's the same as `dev` but using `localhost` for the catalog service.

**Review by petesantos — APPROVED** [review]

**Pull request merged**